plugins/importexport/csv/CSVImportExportPlugin.inc.php

```diff
@@ -82,190 +82,414 @@ function executeCLI($scriptName, &$args) {
+			exit(1);
 		}
 
-		$data = file($filename);
+		$file->setFlags(SplFileObject::READ_CSV);
+
+		if ($file->eof()) {
+			echo __('plugins.importexport.csv.fileDoesNotExist', ['filename' => $filename]) . "\n";
+			exit;
+		}
+
+		$expectedHeaders = [
+			'pressPath',
+			'authorString',
+			'title',
+			'abstract',
+			'seriesPath',
+			'year',
+			'isEditedVolume',
+			'locale',
+			'filename',
+			'doi',
+			'keywords',
+			'subjects',
+			'bookCoverImage',
+			'bookCoverImageAltText',
+			'categories',
+		];
+		$headers = $file->fgetcsv();
+
+		$missingHeaders = array_diff($expectedHeaders, $headers);
+
+		if (count($missingHeaders)) {
+			echo __('pugin.importexport.csv.missingHeadersOnCsv', ['missingHeaders' => $missingHeaders]);
+			exit(1);
+		}
+
+		// Imports
+		import('lib.pkp.classes.submission.SubmissionFile'); // constants.
+		import('lib.pkp.classes.file.FileManager');
+		import('lib.pkp.classes.core.Core');
+		import('classes.file.PublicFileManager');
+
+		// DAOs
+		$pressDao = Application::getContextDAO();
+		$genreDao = DAORegistry::getDAO('GenreDAO'); /* @var $genreDao GenreDAO */
+		$userGroupDao = DAORegistry::getDAO('UserGroupDAO'); /* @var $userGroupDao UserGroupDAO */
+		$submissionDao = DAORegistry::getDAO('SubmissionDAO'); /* @var $submissionDao SubmissionDAO */
+		$publicationDao = DAORegistry::getDAO('PublicationDAO'); /* @var $publicationDao PublicationDAO */
+		$seriesDao = DAORegistry::getDAO('SeriesDAO'); /* @var $seriesDao SeriesDAO */
+		$authorDao = DAORegistry::getDAO('AuthorDAO'); /* @var $authorDao AuthorDAO */
+		$publicationFormatDao = DAORegistry::getDAO('PublicationFormatDAO'); /* @var $publicationFormatDao PublicationFormatDAO */
+		$publicationDateDao = DAORegistry::getDAO('PublicationDateDAO'); /* @var $publicationDateDao PublicationDateDAO */
+		$submissionFileDao = DAORegistry::getDAO('SubmissionFileDAO'); /* @var $submissionFileDao SubmissionFileDAO */
+		$categoryDao = DAORegistry::getDAO('CategoryDAO'); /* @var $categoryDao CategoryDAO */
+
+		// Static variables
+		$fileManager = new FileManager();
+		$publicFileManager = new PublicFileManager();
+		$dirNames = Application::getFileDirectories();
+		$format = trim($dirNames['context'], '/') . '/%d/' . trim($dirNames['submission'], '/') . '/%d';
+		$allowedFileTypes = ['gif', 'jpg', 'png', 'webp'];
+
+		/** @var \PKP\services\PKPFileService */
+		$fileService = Services::get('file');
+		/** @var \PKP\services\PKPPublicationService */
+		$publicationService = Services::get('publication');
+
+		// Cache variables
+		$genres = [];
+		$authorGroups = [];
+		$series = [];
+		$cachedCategories = [];
+		$presses = [];
+
+		$csvForInvalidRowsName = "{$basePath}/invalid_rows.csv";
+		$invalidRowsFile = new SplFileObject($csvForInvalidRowsName, 'w');
+
+		$invalidRowsFile->fputcsv(array_merge(array_pad($headers, 15, null), ['error']));
+
+		foreach ($file as $index => $fields) {
+			if (!$index) {
+				continue; // Skip headers
+			}
```

Comment:
As you're reading the headers on the line 134, this should be removed.

Reply:
On my tests, even reading the headers above, the foreach starts by the file beginning, so the headers are repeated here. I just don't know why :laughing:

Reply:
Hmm, so the iterator is probably resetting the file pointer to the beginning, good to know 😁